lib/sidekiq/instrument/middleware/client.rb

```diff
@@ -10,6 +10,7 @@ def call(worker_class, job, queue, redis_pool)
       class_instance = klass.new
       Statter.statsd.increment(metric_name(class_instance, 'enqueue'))
       Statter.dogstatsd&.increment('sidekiq.enqueue', worker_dog_options(class_instance))
+      Statter.dogstatsd&.flush(sync: true)
```

Comment:
move this line after `yield`, so that it can also flush any custom metrics in the worker.